internal/bminventory/inventory.go

```diff
@@ -3255,13 +3256,40 @@ func (b *bareMetalInventory) RegisterOCPCluster(ctx context.Context) error {
 		return err
 	}
 
+	txSuccess := false
```

Comment:
you can use `b.db.Transaction(func(tx *gorm.DB) error` and then you will not need the defer part

Reply:
yes, but i think that transaction code is too long for inline - to me it looks that the function is more readable this way.

Reply:
I don't think it's a good idea - we don't want to encourage writing functions that affect states outside of a state machine

Reply:
sorry wrong comment, ignore.

Reply:
I'm not sure that i understand your reply. you already create a transaction so how will it make the code longer?

Reply:
unless your comment is related to `RegisterInstalledOCPHost`?